nautobot/ipam/querysets.py

```diff
@@ -327,52 +327,38 @@ def _validate_cidr(self, value):
         except netaddr.AddrFormatError as err:
             raise ValidationError({"cidr": f"{value} does not appear to be an IPv4 or IPv6 network."}) from err
 
-    def get_closest_parent(self, cidr, namespace, max_prefix_length=0):
```

Comment:
Refactored `get_closest_parent` with @jathanism to fix case where /32 IPv4 address was not matching /32 prefix